File: Services/java-services/Core/CoreShared/src/main/java/ai/omnicure/core/shared/pagination/PagedParams.java
package ai.omnicure.core.shared.pagination;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class PagedParams {

    private int pageNumber = 1;
    private int pageSize = 10;
    private String searchKeyword;
    private String sortBy;
    private String sortDirection = "asc";

    public void setPageSize(int pageSize) {
        if (pageSize > 1000) {
            this.pageSize = 1000;
        } else if (pageSize < 1) {
            this.pageSize = 1;
        } else {
            this.pageSize = pageSize;
        }
    }

    public void setPageNumber(int pageNumber) {
        this.pageNumber = Math.max(pageNumber, 1);
    }
}

File: Services/dotnet-services/Core/OmniCure.Core.Shared/Pagination/PagedParams.cs
namespace OmniCure.Core.Shared.Pagination;

public class PagedParams
{
    private int _pageSize = 10;

    public int PageNumber { get; set; } = 1;

    public int PageSize
    {
        get => _pageSize;
        set => _pageSize = value switch
        {
            > 1000 => 1000,
            < 1 => 1,
            _ => value
        };
    }

    public string? SearchKeyword { get; set; }

    public string? SortBy { get; set; } = default!;

    public string? SortDirection { get; set; } = "asc";
}
